src/store/model-store.js

```diff
@@ -1,33 +1,26 @@
 import { Store } from 'consus-core/flux';
-
+import { searchModel } from '../lib/api-client';
 let model = null;
 let models = [];
 
 class ModelStore extends Store {
 
     getModel() {
-        if (model === null) {
-            return model;
-        }
-        return {
-            address: model.address,
-            name: model.name
-        };
+        return model;
     }
 
     getAllModels(){
         return models;
     }
-
+    searchModelByAddress(address){
+        return searchModel(address);
+    }
 }
 
 const store = new ModelStore();
 
 store.registerHandler('MODEL_FOUND', data => {
-    model = {
-        address: data.address,
-        name: data.name
-    };
+    model = data;
```

Comment:
See previous comment about repeated code and checking the api documents for the attributes. 

TLDR; I hid away the attributes in the client and you need to look for attributes in the api documents for consus to prevent attributes missing from server and client because we forgot to explicitly define them.